### ISSUE TITLE
[Question] typeorm CLI migration:run file order

### ISSUE THREAD

**Issue by StickNitro** [typeorm/typeorm#1250]
I have a project which I am using typeorm in and I have my migrations generating when I make schema changes however, when I run `typeorm migrations:run` on the terminal the migrations files are executed in the wrong order.  I currently have four migrations, these are:

```
1511875872426-Initial.ts
1511875989237-InsertDefaultUser.ts
1511876128735-InsertStrengths.ts
1511876181411-InsertStrengthsItems.ts
```

The above is the order that they appear in the folder tree in vscode. but when I run migrations it seems to want to run `1511875989237-InsertDefaultUser.ts` first but of course the table does not yet exist since `1511875872426-Initial.ts` has not yet been run.

Am I doing something wrong, please advise?

**Comment by pleerock** [comment]
I just checked it and they are executed in proper order first 1511875872426-Initial then 1511875989237-InsertDefaultUser, can you provide a minimal reproduction git repo?

**Comment by pleerock** [comment]
Closing as no answer from reporter.

**Comment by ethanorlanderypm** [comment]
I experienced the same issue and found the following cause and resolution:

TL;DR: The order the migrations run in is determined by the ISO 8601 time stamp at the end of the `name` attribute in a migration's class (which is the migration file), _not_ the time stamp in the class name, or as you might (wrongly) assume, the timestamp prepending the filename.

For example, let's say we have two migration files:

1591832825936-CreateDatabase.ts
1591832847383-SomeMigration.ts

The time stamp on the CreateDatabase migration is less than that of SomeMigrations, so we might assume it would run first. However, the first few lines of the files are as such:

1591832825936-CreateDatabase.ts:
```
import { MigrationInterface, QueryRunner } from "typeorm";

export class CreateDatabase1591832887383 implements MigrationInterface {
	name = "CreateDatabase1591832887383";
```

1591832847383-SomeMigration.ts
```
import { MigrationInterface, QueryRunner } from "typeorm";

export class SomeMigration1591832825936 implements MigrationInterface {
	name = "SomeMigration1591832825936";
```

In the CreateDatabase file, the timestamp in the attribute `name = "CreateDatabase1591832887383";` does not match the timestamp at the beginning of the file name _1591832825936-CreateDatabase.ts_. In fact, the time at the end of the `name` attribute in CreateDatabase is actually greater than the time on the `name` attribute in SomeMigration, and because of this, SomeMigration will run first, even though the time on the filenames might lead you to assume otherwise.

**Comment by sadi304** [comment]
@ethanorlanderypm  good find. This should be mentioned in docs I think

**Comment by remes2000** [comment]
Also, keep in mind that if you'll have an empty space at the end of the name property:
```
export class SomeMigration1728055819056 implements MigrationInterface {
  name = 'SomeMigration1728055819056 ';
...
```
TypeORM is going to run this migration file as first :D Silly mistake form my side, but it was very painful to find out the cause
